// plugin-welcome/src/api/models/EntPatientrecordEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntHistorytaking,
    EntHistorytakingFromJSON,
    EntHistorytakingFromJSONTyped,
    EntHistorytakingToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPatientrecordEdges
 */
export interface EntPatientrecordEdges {
    /**
     * Historytaking holds the value of the historytaking edge.
     * @type {Array<EntHistorytaking>}
     * @memberof EntPatientrecordEdges
     */
    historytaking?: Array<EntHistorytaking>;
}

export function EntPatientrecordEdgesFromJSON(json: any): EntPatientrecordEdges {
    return EntPatientrecordEdgesFromJSONTyped(json, false);
}

export function EntPatientrecordEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPatientrecordEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'historytaking': !exists(json, 'historytaking') ? undefined : ((json['historytaking'] as Array<any>).map(EntHistorytakingFromJSON)),
    };
}

export function EntPatientrecordEdgesToJSON(value?: EntPatientrecordEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'historytaking': value.historytaking === undefined ? undefined : ((value.historytaking as Array<any>).map(EntHistorytakingToJSON)),
    };
}
